cmake/UseCcache.cmake

```
@@ -0,0 +1,59 @@
+# Compiling ccache via ccache is fine, because this uses a stable version which
+# is installed on the system.
+
+# Modified version of Craig Scott's "Professional CMake: A Practical Guide", 7th Edition
+function(use_ccache)
+  if(NOT CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    message(STATUS " use_ccache() disabled, as it is not called from the project top level")
```

The status messages still look a bit like debug messages to me: They contain an extra space prefix and some of them mention internal details like function name which is not relevant for a normal project user.

Idea: Add a `MESSAGE_DEBUG` function that calls `message(DEBUG ...)` for CMake ≥ 3.15 and nothing for older versions. Then:

```suggestion
    MESSAGE_DEBUG("use_ccache() disabled, as it is not called from the project top level")
```

What do you think?

---

I made them VERBOSE. See https://cmake.org/cmake/help/latest/command/message.html

---

OK, but neither VERBOSE nor DEBUG exist for CMake < 3.15 so they will be always printed with a strange literal "VERBOSE" prefix for those versions.

---

Argh